### PR TITLE
Add a proxy delay to checklist parameter changes via children edits

### PR DESCRIPTION
**Pull request by ntjess**
Randomly testing with a touchpad I found 1s proxy delay was usually adequate. Input from others would be nice on this

https://user-images.githubusercontent.com/23620506/141720037-ab4d8c2f-80d5-4cd8-8751-8f98d662f097.mp4

.

### CODE REVIEW

**Comment by ntjess** [comment]
This is similar to how the SpinBox parameter (int/float) doesn't emit value changes until after the delay is reached. I just wasn't sure about hard-coding 1s delay, so I just made it an option.

**Comment by j9ac9k** [comment]
Hi @ntjess I haven't worked w/ this much, but what is the issue that this solves?

**Comment by ntjess** [comment]
Same issue that is solved in spinbox. It lets individual check boxes act as sigValueChanging while valueChanged only fires when the user is done editing

**Comment by j9ac9k** [comment]
Oh, so in this case, when you're checking the checkboxes, instead of running the function that's being called on each time it's checked _as_ its checked, it waits a second to see if a user checks anything else, and then runs the slot that's connected to the `stateChanged` signal (in the case of the checkboxes)?

**Comment by ntjess** [comment]
Yep! Just like changing vs changed on a spinbox (waiting until the spinner is motionless for awhile)

File: pyqtgraph/parametertree/parameterTypes/checklist.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from ... import functions as fn
-from ...Qt import QtWidgets
+from ...Qt import QtWidgets, QtCore
```
Comment (j9ac9k) [inline pyqtgraph/parametertree/parameterTypes/checklist.py:2]: Can we get rid of the QtCore import here?

**Comment by j9ac9k** [comment]
LGTM, thanks @ntjess for the PR!

**Comment by ntjess** [comment]
Looking at https://github.com/pyqtgraph/pyqtgraph/pull/2086#issuecomment-972824641, should this logic be changed to "when no child is focused" rather than "after 1 second"?

**Comment by j9ac9k** [comment]
I can see a case for being made for if lose focus, emit signal right away, else wait for the delay period.

**Pull request merged**